pushy/src/main/java/com/turo/pushy/apns/server/UnregisteredDeviceTokenException.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2013-2017 Turo
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.turo.pushy.apns.server;
+
+import java.util.Date;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * An exception thrown by {@link PushNotificationHandler} instances to indicate that a push notification should be
+ * rejected by the server because its destination device token is no longer valid (presumably because the receiving app
+ * has been removed from the destination device).
+ *
+ * @since 0.12
+ */
+public class UnregisteredDeviceTokenException extends RejectedNotificationException {
```

Comment:
Felt weird to have `getDeviceTokenExpirationTimestamp` in the base `RejectedNotificationException` if only one error code could reasonably use it. Feels weird to have a special subclass for a single error code. I guess the alternative would be to have an exception for each kind of rejection, but that feels like overkill.